tests/unit/test_update_issue.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2025 Canonical Ltd.
```

Comment:
Note that I'm not intending to get complete or even high test coverage for this file, just the "most testable" pieces covered.